pkg/store/cache/cache.go

```diff
@@ -0,0 +1,325 @@
+package storecache
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	lru "github.com/hashicorp/golang-lru/simplelru"
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/tsdb/labels"
+)
+
+const (
+	cacheTypePostings string = "Postings"
+	cacheTypeSeries   string = "Series"
+
+	sliceHeaderSize = 16
+)
+
+type cacheItem struct {
```

Comment:
Reading the code is made somewhat more confusing by this being called `cacheItem` when it is used as the cache key. WDYT about renaming to `cacheKey`?

Reply:
yes!